great_expectations/data_context/types/base.py

```diff
@@ -254,6 +254,12 @@ class Meta:
     schema_name = fields.String(required=False, allow_none=True)
     batch_spec_passthrough = fields.Dict(required=False, allow_none=True)
 
+    # Necessary addition for AWS Glue Data Catalog assets
+    database_name = fields.String(required=False, allow_none=True)
+    partitions = fields.List(
+        cls_or_instance=fields.Str(), required=False, allow_none=True
+    )
+
```

Comment:
@lccasagrande Question: Do we need these fields at the data connector level as well?  The use case would be specify at the data connector the general defaults and override them with the specifics at the `Asset` level.  I am doing that in #5847 (under review).  Thanks!

Reply:
Not sure if I got what you mean. I have added this to allow the user to specify just a small set of partitions, instead of using all table partitions. 

Example:
Lets suppose that Table A has three partitions: year, month, and day. If you do not set the partitions in the config, the connector will create just one batch id that has data for all three partitions in it. If you set partitions to year and day, glue will create N batch ids with the combination of both partitions. In this case, you would have one batch per combination of year and month and you could filter the batch you want to validate.

I do not believe that these fields should be at the data connector level because each asset (table) can have different partition keys.

Reply:
@lccasagrande But sometimes if you have all tables that are similar and their partition keys can be the same -- it is easier to specify them at the data connector level.  If a specific override is needed at an asset level, it can be provided.  Would that make sense?  This would mirror what is done in the existing `SQL` connector / asset code.  Thanks!

Reply:
Done